### ISSUE TITLE
Port assignments

### ISSUE THREAD

**Issue by bpbond** [PNNL-TES/kpsr#3]
Hi @kaizadp 

So the data start with a long (two day) read on port 0, and there's another very long one on port 4 (see graphs below). Is this intended? 

![k1](https://user-images.githubusercontent.com/1956468/57944692-cc0f4300-78a5-11e9-97ba-9611411c7279.png)
![k2](https://user-images.githubusercontent.com/1956468/57944693-cc0f4300-78a5-11e9-9e4b-3333e6ce1c26.png)

**Comment by kaizadp** [comment]
Hi @bpbond -- no, those were not intended, and I'm not sure what port 0 is. Is that background when no program is running? 

I started my experiment on 5/2. Ran just a couple of ports initially, and then began using ports 1-5 and ambient port 16 around 5/2, 7:56 pm, which was 5/2 8:47 pm on the Picarro computer, but 5/3 1:47 am on the Picarro software. I wasn't sure how all these related, and I also wanted to see if/how the R-code would pull the required subset out, so I left it all in. In the valvemap file, I set the start_date_time to 5/2/19, 7:56 pm.
Is that what's messing it up? I can go in and remove the long reads on 0 and 4.

**Comment by bpbond** [comment]
OK, thanks, that helps. I'll come back to this later.

>I can go in and remove the long reads on 0 and 4.

No, no! Raw data is sacred. I just want to make sure I understand what's going on, and then we can handle it correctly.

**Comment by bpbond** [comment]
So there are 6,002 Picarro observations, but after filtering the dataset to drop ambient (valve 16) data, and limiting to first two minutes of data for each port, this drops to 383. Visualized by port and elapsed time:

![plot](https://user-images.githubusercontent.com/1956468/58098305-42ae8800-7ba7-11e9-9d42-8a040361c311.png)

So we want to focus just on ports 1-5 per above?

**Comment by kaizadp** [comment]
That sounds about right. Yes, ports 1-5. Thanks!

**Comment by bpbond** [comment]
Another thing I'm seeing is relatively few measurements per sample. Just out of curiosity, did you change the Picarro settings in this regard? In most cases we only have 3-4 points in the first two minutes (see above), which is pretty much the minimum for computing a flux.

**Comment by bpbond** [comment]
Changing the max measurement time from 45 to 90 seconds gives us more data to work with, but still not much:

```
> summarydata
# A tibble: 12 x 10
   samplenum CO2_ppm_s CH4_ppb_s    r2_CO2    r2_CH4 max_co2_time DATETIME                N MPVPosition h2o_reported
       <int>     <dbl>     <dbl>     <dbl>     <dbl>        <dbl> <dttm>              <int>       <dbl>        <dbl>
 1         8    0.0892  -0.00623    0.816     0.816            83 2019-05-02 23:32:10     4           3        2.26 
 2        14   NA       NA          0         0                27 2019-05-03 00:01:57     1           3        0.718
 3        16    2.92   -11.1      NaN       NaN                49 2019-05-03 00:14:36     2           1        2.29 
 4        29    0.265    6.84       0.656     0.656            76 2019-05-03 01:27:46     3           1        0.705
 5        32    1.10   -11.9      NaN       NaN                47 2019-05-03 01:42:32     2           1        0.701
 6        46   NA       NA          0         0                29 2019-05-03 02:42:08     1           1        0.697
 7        60   NA       NA          0         0                 6 2019-05-03 03:06:53     1           3        2.52 
 8        95    0.612   -0.0372    -0.279    -0.279            51 2019-05-03 04:05:25     3           2        1.42 
 9       105   NA       NA          0         0                49 2019-05-03 04:20:46     1           3        2.79 
10       111   NA       NA          0         0                28 2019-05-03 04:29:14     1           1        2.31 
11       117   NA       NA          0         0                28 2019-05-03 04:42:19     1           1        2.31 
12       137    0.0297  -0.0250    -0.0791   -0.0791           54 2019-05-03 20:01:01     3           3        0.833
```
This was a test run I hope?

**Comment by kaizadp** [comment]
I set it to measure for 2 minutes, but I didn’t change the sampling interval within that. There is ambient measurement after each core, and I reduced that to 0.5 minute. It looks like the Picarro is measuring every 25-50 seconds, but not consistently. Is that normal? I could adjust those parameters next time.

**Comment by kaizadp** [comment]
Yes, this was a test run! I wanted to be sure I understood how to use the Picarro and the code before I began an experiment. 

So we filter out the measurements with only 2 points. Is that why we have so few flux values in your table?

And I didn’t realize the measurement time was previously set to 45 seconds. That means I’ll have to set the Picarro to measure more frequently next time.

**Comment by kaizadp** [comment]
Yes, this was a test run! I wanted to be sure I understood how to use the Picarro and the code before I began an experiment.

So we filter out the measurements with only 2 points. Is that why we have so few values in your summary table?

And I didn’t realize the measurement time was previously set to 45 seconds. That means I’ll have to set the Picarro to measure more frequently next time

**Comment by bpbond** [comment]
I have opened a "pull request" with script changes--this makes it easy for you to review and see what I changed. If it looks good, click "Merge".

https://github.com/PNNL-TES/kpsr/pull/4
